### PR TITLE
Update for RabbitMQ 3.13

### LINKED ISSUE #263: Support for RabbitMQ 3.13.x
### Is your feature request related to a problem? Please describe.

The rabbitmq-delayed-message-exchange plugin does not list 3.13 as supported version yet.
Version 3.13 of RabbitMQ-Server released recently.
Running e.g. `apt-upgrade` would lead to problems / incompatibility.

### Describe the solution you'd like

Please add version 3.13.x to the list of supported version.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #263

### CODE REVIEW

**Review by pjk25 — COMMENTED** [review]
I think we would want this to follow the 3.13.x branch instead of main?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,8 +13,8 @@ jobs:
       fail-fast: false
       matrix:
         bazel_otp_name:
-        - "25"
-        - "26"
+          - "25"
+          - "26"
```
Comment (pjk25) [inline .github/workflows/test.yml:17]: I think this needs to be `25_3` and `26_2`

Comment (lukebakken) [reply .github/workflows/test.yml:17]: Thanks!

**Comment by lukebakken** [comment]
> I think we would want this to follow the 3.13.x branch instead of main?

@pjk25 My plan is to merge this to `main`, then immediately create a `v3.13.x` branch on that commit.

I could just rename `main` to `v3.13.x`, too. Thoughts?

cc @michaelklishin

**Comment by michaelklishin** [comment]
Yup, just like with other branches and plugins.

**Comment by pjk25** [comment]
The tests fail, something khepri related https://app.buildbuddy.io/invocation/99857d85-5002-453d-a78a-cbee7e49957d?target=%2F%2F%3Aplugin_SUITE&targetStatus=6

**Comment by pjk25** [comment]
> > I think we would want this to follow the 3.13.x branch instead of main?
> 
> @pjk25 My plan is to merge this to `main`, then immediately create a `v3.13.x` branch on that commit.
> 
> I could just rename `main` to `v3.13.x`, too. Thoughts?
> 
> cc @michaelklishin

I meant this change:
```diff --git a/MODULE.bazel b/MODULE.bazel
index 17a2196..d00f919 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -32,9 +32,9 @@ bazel_dep(

 archive_override(
     module_name = "rabbitmq-server",
-    strip_prefix = "rabbitmq-server-main",
+    strip_prefix = "rabbitmq-server-3.13.x",
     urls = [
-        "https://github.com/rabbitmq/rabbitmq-server/archive/refs/heads/main.zip",
+        "https://github.com/rabbitmq/rabbitmq-server/archive/refs/heads/v3.13.x.zip",
     ],
 )
```

Probably best to create the v3.13.x branch on this repo, merge this PR to it, then have a separate PR for main. Other option is to make the above diff part of this PR, but then revert it on main of this repo

**Comment by lukebakken** [comment]
@pjk25 - I think this PR represents what you suggested. Merge to `v3.13.x`, different PR for `main`

**Review by pjk25 — APPROVED** [review]

**Pull request merged**